client/src/app/core/core-services/http-stream.service.ts

```diff
@@ -14,24 +15,33 @@ export class StreamConnectionError extends Error {
     }
 }
 
-export class StreamContainer {
+export class StreamContainer<T> {
     public readonly id = Math.floor(Math.random() * (900000 - 1) + 100000); // [100000, 999999]
 
     public retries = 0;
     public hasErroredAmount = 0;
-    public errorHandler: (error: any) => void = null;
+    public errorHandler: (type: ErrorType, error: CommunicationError, message: string) => void = null;
 
-    public messageHandler: (message: any) => void;
-    public stream?: Stream<any>;
+    public messageHandler: (message: T) => void;
+    public stream?: Stream<T>;
 
     public constructor(
         public endpoint: EndpointConfiguration,
-        messageHandler: (message: any, streamId: number) => void,
+        messageHandler: (message: T, streamId: number) => void,
         public params: () => Params,
         public body: () => any,
         public description: string
     ) {
-        this.messageHandler = (message: any) => messageHandler(message, this.id);
+        // this.messageHandler = (message: T) => messageHandler(message, this.id);
+        this.messageHandler = (message: T) => {
+            // {connected: true} is a special message just to trigger the code below
```

Comment:
Please change this to the following code since we do not have connect messages anymore
```
if (this.hasErroredAmount  > 0) {
    console.log(`resetting error amount for ${this.endpoint} since there was a connect message`);                
    this.hasErroredAmount = 0;
}
messageHandler(message, this.id);
```